coil-core/src/androidMain/kotlin/coil3/decode/FastImageDecoderDecoder.kt

```diff
@@ -0,0 +1,194 @@
+package coil3.decode
+
+import android.graphics.ImageDecoder
+import android.graphics.drawable.AnimatedImageDrawable
+import android.graphics.drawable.Drawable
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.core.util.component1
+import androidx.core.util.component2
+import coil3.Image
+import coil3.ImageLoader
+import coil3.asCoilImage
+import coil3.drawable.ScaleDrawable
+import coil3.fetch.SourceFetchResult
+import coil3.request.Options
+import coil3.request.allowRgb565
+import coil3.request.animatedTransformation
+import coil3.request.animationEndCallback
+import coil3.request.animationStartCallback
+import coil3.request.bitmapConfig
+import coil3.request.colorSpace
+import coil3.request.premultipliedAlpha
+import coil3.request.repeatCount
+import coil3.toAndroidUri
+import coil3.util.animatable2CallbackOf
+import coil3.util.asPostProcessor
+import coil3.util.heightPx
+import coil3.util.isHardware
+import coil3.util.widthPx
+import kotlin.math.roundToInt
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.sync.Semaphore
+import kotlinx.coroutines.sync.withPermit
+import kotlinx.coroutines.withContext
+
+@RequiresApi(28)
+class FastImageDecoderDecoder(
```

Comment:
Let's name this `StaticImageDecoderDecoder` and the one in `coil-gif` can be `AnimatedImageDecoderDecoder`. Ideally `StaticImageDecoder` will decode all images as a bitmap and `AnimatedImageDecoderDecoder` will support decoding animated images as fully animated.

This way we have consistent behaviour across all API levels:

- `coil-core` only decodes static images
- `coil-gif`'s decoders add support for animated images

Reply:
Then we only use ImageDecoder.decodeBitmap for StaticImageDecoderDecoder?
SGTM

Reply:
Done